libbeat/processors/add_kubernetes_metadata/config.go

```diff
@@ -43,8 +46,52 @@ type Enabled struct {
 
 type PluginConfig []map[string]common.Config
 
+func trimAfter(value string, a string) string {
+	// Get substring after a string.
+	pos := strings.LastIndex(value, a)
+	if pos == -1 {
+		return ""
+	}
+	adjustedPos := pos + len(a)
+	if adjustedPos >= len(value) {
+		return ""
+	}
+	return value[adjustedPos:len(value)]
+}
+
+func findDaemonsConfigFlag() string {
```

Comment:
I would remove the `ps aux`, sounds like an unexpected trick. While it is doing its best to find kubeconfigs, it is also a _too unique way_ of doing it. This could lead to unexpected behaviours at clients.

IMO, `findDaemonsConfigFlag` is not needed, unless there is a case you are thinking of. For now I'd recommend to remove it from this PR to proceed.